api/apps/geoprocessing/src/export/pieces-exporters/project-metadata.ts

```diff
@@ -1,19 +1,67 @@
 import { Injectable } from '@nestjs/common';
+import { InjectEntityManager } from '@nestjs/typeorm';
+import { EntityManager } from 'typeorm';
+import { Readable } from 'stream';
+
 import { ClonePiece, JobInput, JobOutput } from '@marxan/cloning';
+import { FileRepository } from '@marxan/files-repository';
+
+import { geoprocessingConnections } from '@marxan-geoprocessing/ormconfig';
+
 import { PieceExportProvider, PieceProcessor } from '../pieces/piece-processor';
+import { isLeft } from 'fp-ts/Either';
 
 @Injectable()
 @PieceExportProvider()
 export class ProjectMetadata extends PieceProcessor {
+  constructor(
+    private readonly fileRepository: FileRepository,
+    @InjectEntityManager(geoprocessingConnections.apiDB)
+    private readonly entityManager: EntityManager,
+  ) {
+    super();
+  }
+
   isSupported(piece: ClonePiece): boolean {
     return piece === ClonePiece.ProjectMetadata;
   }
 
   async run(input: JobInput): Promise<JobOutput> {
-    console.log(`exporting project metadata`, input);
+    const projectData: Array<{
+      name: string;
+      description: string;
+    }> = await this.entityManager.query(
+      `
+    SELECT projects.name, projects.description FROM projects WHERE projects.id = $1
+    `,
+      [input.resourceId],
+    );
+
+    if (projectData.length !== 1) {
+      throw new Error(
+        `${ProjectMetadata.name} - Project ${input.resourceId} does not exist.`,
+      );
+    }
+
+    const metadata = JSON.stringify({
+      name: projectData[0].name ?? null,
+      description: projectData[0].description ?? null,
+    });
```

Comment:
very minor since this is just a metadata exporter, but shouldn't we let it fail if there's no name? given the data model, this shouldn't happen, I think 🤔

Reply:
right, `name` is required. Fixed in https://github.com/Vizzuality/marxan-cloud/pull/686/commits/2dc2f385d2b4c2c3c14e9b5a428ac0f490e5a5e9